github/downloader_test.go

```diff
@@ -0,0 +1,205 @@
+package github
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/src-d/metadata-retrieval/github/store"
+	"github.com/stretchr/testify/require"
+
+	"golang.org/x/oauth2"
+)
+
+func TestOnlineRepositoryDownload(t *testing.T) {
+	var tests = []struct {
+		owner           string
+		repository      string
+		version         int
+		url             string
+		createdAt       string
+		isPrivate       bool
+		isArchived      bool
+		hasWiki         bool
+		numOfPrs        int
+		numOfPrComments int
+	}{
+		{
+			"git-fixtures",
+			"basic",
+			0,
+			"https://github.com/git-fixtures/basic",
+			"2015-03-31 11:42:21 +0000 UTC",
+			false,
+			false,
+			false,
+			2,
+			4,
+		},
+		{
+			"git-fixtures",
+			"symlinks",
+			0,
+			"https://github.com/git-fixtures/symlinks",
+			"2017-06-18 17:57:27 +0000 UTC",
+			false,
+			false,
+			true,
+			0,
+			0,
+		},
+		{
+			"git-fixtures",
+			"submodule",
+			0,
+			"https://github.com/git-fixtures/submodule",
+			"2017-02-12 17:18:40 +0000 UTC",
+			false,
+			false,
+			true,
+			0,
+			0,
+		},
+		{
+			"git-fixtures",
+			"releases",
+			0,
+			"https://github.com/git-fixtures/releases",
+			"2016-11-06 21:27:46 +0000 UTC",
+			false,
+			false,
+			true,
+			0,
+			0,
+		},
+		{
+			"git-fixtures",
+			"tags",
+			0,
+			"https://github.com/git-fixtures/tags",
+			"2016-09-21 19:15:12 +0000 UTC",
+			false,
+			false,
+			true,
+			0,
+			0,
+		},
+		{
+			"git-fixtures",
+			"empty",
+			0,
+			"https://github.com/git-fixtures/empty",
+			"2016-08-25 12:48:01 +0000 UTC",
+			false,
+			false,
+			true,
+			0,
+			0,
+		},
+		{
+			"git-fixtures",
+			"small-several-contributors",
+			0,
+			"https://github.com/git-fixtures/small-several-contributors",
+			"2016-02-23 18:58:59 +0000 UTC",
+			false,
+			false,
+			true,
+			0,
+			0,
+		},
+		{
+			"git-fixtures",
+			"small-several-contributors-with-prs",
+			0,
+			"https://github.com/git-fixtures/small-several-contributors-with-prs",
+			"2016-02-23 19:20:39 +0000 UTC",
+			false,
+			false,
+			true,
+			3,
+			1,
+		},
+	}
+
+	client := oauth2.NewClient(context.TODO(), oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: os.Getenv("GITHUB_TOKEN")},
+	))
+
+	var downloader, err = NewStdoutDownloader(client)
+
+	if err != nil {
+		t.Errorf("Downloader failed to init with error:%s", err)
+	}
+
+	saved := downloader.storer
+	defer func() { downloader.storer = saved }()
+	var teststore *store.TestStore = &store.TestStore{}
+	downloader.storer = teststore
+
+	for _, test := range tests {
```

Comment:
it's a good idea to use `t.Run` for table tests usually.
also if you are tired of passing `t` everywhere `require := require.New(t)` can help with it.